### PR TITLE
Phase field fracture displacement vector and c > 1.0 fix

### LINKED ISSUE #8197: PF fracture model issue: c goes above 1
### Description of the enhancement or error report
In the phase field fracture model, we usually want to input displacement as a vector instead of two or three scalars, so we modified the corresponding codes to input the displacement as a vector input.
Also, when we run the simulation, the c goes above 1. 
### Rationale for the enhancement or information for reproducing the error
Usually c should be less than or equal to 1, and stress is a function of c and goes to minimum value when c=1, also the stress would increase when c is larger than 1. Thus, it is necessary for us to fix this issue to get the correct stress distribution. In order to fix this issue, we added one new parameter p=1-c in the kernel code file in which p=0 when c goes above 1, this could make sure that the stress would not increase when c keeps going above 1.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

### PR DESCRIPTION
**Pull request by dschwen**
Ping @zhangsf2015 

Replaces #8206
Closes #8197

### CODE REVIEW

**Comment by dschwen** [comment]
Ha, my commit breaks this PR. I'll fix it tomorrow.

**Comment by dschwen** [comment]
The two exodiffs are due to me switching the code to use `MathUtils::sign` (which returns +1 for an argument of 0) instead of the ternary operator (which returns -1 for 0).

The crashed test requires further investigation...

**Comment by dschwen** [comment]
Yeah, the happens without my commit as well.

**Comment by dschwen** [comment]
Odd, those tests pass on my mac (but there the grain_tracker_volume test crashes).

File: modules/phase_field/src/kernels/PFFracIntVar.C [hunk 1]
```diff
@@ -23,14 +23,12 @@ PFFracIntVar::PFFracIntVar(const InputParameters & parameters):
...
+  // Residual is the variable value
   return _u[_qp];
 }
 
 Real
 PFFracIntVar::precomputeQpJacobian()
 {
-  Real val=1.0;
-  return val * _phi[_j][_qp];
+  return _phi[_j][_qp];
```
Comment (tonkmr) [inline modules/phase_field/src/kernels/PFFracIntVar.C]: This is identical to the reaction kernel. We should deprecate this kernel.

Comment (dschwen) [inline modules/phase_field/src/kernels/PFFracIntVar.C]: Agree. Once I figure out where the diffing comes from (mac/linux discrepancy) I'll amend this PR.

**Comment by tonkmr** [comment]
This looks good to me

**Comment by tonkmr** [comment]
@dschwen I just discovered another repeated kernel here. PFFracCoupledInterface is exactly the same as CHSplitVar

**Review by zhangsf2015 — APPROVED** [review]
Hi Daniel,
Happy new year. Sorry for the late reply and thanks for your help. It looks great  for me, and I approve the changes.

**Review by permcody — APPROVED** [review]
Approved by @zhangsf2015

**Comment by dschwen** [comment]
@tonkmr I'll make a new PR for the PFFracCoupledInterface -> CHSplitVar deprecation. This one is too far along.

**Comment by permcody** [comment]
@dschwen - please resolve conflicts

**Comment by dschwen** [comment]
@permcody conflicts are resolved

**Pull request merged**